packages/sdk/src/core/integration/IntegrationClient.ts

```diff
@@ -0,0 +1,325 @@
+    props: GetConnectionPipelinesRequest & { enablePagination?: boolean },
+  ) {
+    const {
+      pageSize,
+      pageToken,
+      enablePagination,
+      filter,
+      namespaceId,
+      connectionId,
+    } = props;
+
+    try {
+      const pipelineIds: string[] = [];
+
+      const queryString = getQueryString({
+        baseURL: `/namespaces/${namespaceId}/connections/${connectionId}/referenced-pipelines`,
+        pageSize,
+        pageToken,
+        filter,
+      });
+
+      const data =
+        await this._client.get<GetConnectionPipelinesResponse>(queryString);
+
+      if (enablePagination) {
+        return Promise.resolve(data);
+      }
+
+      pipelineIds.push(...data.pipelineIds);
+
+      if (data.nextPageToken) {
+        pipelineIds.push(
+          ...(await this.getConnectionPipelines({
+            connectionId,
+            namespaceId,
+            pageSize,
+            pageToken: data.nextPageToken,
+            enablePagination: false,
+            filter,
+          })),
+        );
+      }
+
+      return Promise.resolve(pipelineIds);
+    } catch (error) {
+      return Promise.reject(error);
+    }
+  }
+
+  /**
+   * Gets a specific integration connection
+   */
+  async getIntegration(props: GetIntegrationRequest) {
+    const { view, integrationId } = props;
+
+    try {
+      const queryString = getQueryString({
+        baseURL: `/integrations/${integrationId}`,
+        view,
+      });
+
+      const data = await this._client.get<GetIntegrationResponse>(queryString);
+
+      return Promise.resolve(data.integration);
+    } catch (error) {
+      return Promise.reject(error);
+    }
+  }
+
+  /**
+   * Gets a specific integration
+   */
+  async getIntegrationConnection(props: GetIntegrationConnectionRequest) {
+    const { view, connectionId, namespaceId } = props;
+
+    try {
+      const queryString = getQueryString({
+        baseURL: `/namespaces/${namespaceId}/connections/${connectionId}`,
+        view,
+      });
+
+      const data =
+        await this._client.get<GetIntegrationConnectionResponse>(queryString);
+
+      return Promise.resolve(data.connection);
+    } catch (error) {
+      return Promise.reject(error);
+    }
+  }
+
+  /**
+   * Adds an integration connection
+   */
+  async addIntegration({
+    id,
+    integrationId,
+    namespaceId,
+    setup,
+    method,
+  }: AddIntegrationRequest) {
+    try {
+      const data = await this._client.post<AddIntegrationResponse>(
+        `/namespaces/${namespaceId}/connections`,
+        {
+          body: JSON.stringify({
+            id,
+            integrationId,
+            method,
+            setup,
+          }),
+        },
+      );
+
+      return Promise.resolve(data.connection);
+    } catch (error) {
+      return Promise.reject(error);
+    }
+  }
+
+  /**
+   * Updates an integration connection
+   */
+  async updateIntegrationConnection(props: UpdateIntegrationConnectionRequest) {
```

Comment:
@orangecoloured Let's directly deconstruct the props here since we didn't use it as a whole afterward